src/Frontend/FluentCMS.Web.UI/Components/PluginContainerActions.razor.cs

```diff
@@ -5,41 +5,76 @@ public partial class PluginContainerActions
     [Parameter]
     public PluginViewState Plugin { get; set; } = default!;
 
+    [Parameter]
+    public EventCallback OnUpdate { get; set; } = default!;
+
     [Inject]
-    public ViewState ViewState { get; set; } = default!;
+    private PluginLoader PluginLoader { get; set; } = default!;
 
     [Inject]
-    public NavigationManager NavigationManager { get; set; } = default!;
+    public ViewState ViewState { get; set; } = default!;
 
     [Inject]
     private ApiClientFactory ApiClients { get; set; } = default!;
 
-    private void OpenPluginView(string viewName = "Settings")
+    #region Plugin Edit
+    private bool EditModalOpen { get; set; } = false;
+
+    private Type? GetPluginType(string typeName)
     {
-        var baseUrl = new Uri(NavigationManager.Uri).LocalPath;
+        PluginDefinitionTypeViewState? pluginDefType;
 
-        var queryParams = new List<string>
-        {
-            $"pluginId={Plugin.Id}"
-        };
+        pluginDefType = Plugin.Definition?.Types?.Where(p => p!.Name!.Equals(typeName, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
 
-        if (!string.IsNullOrEmpty(viewName))
-            queryParams.Add($"viewName={viewName}");
+        if (pluginDefType is null)
+            throw new InvalidOperationException("Plugin definition type not found!");
 
-        var redirectTo = Uri.EscapeDataString(baseUrl);
-        if (ViewState.Type == ViewStateType.PagePreview)
-        {
-            redirectTo += Uri.EscapeDataString("?pagePreview=true");
-        }
+        var assemblyName = Plugin?.Definition?.Assembly;
+        if (string.IsNullOrEmpty(assemblyName))
+            throw new InvalidOperationException("Plugin assembly name not found!");
+
+        var type = PluginLoader.GetType(assemblyName, pluginDefType!.Type!) ??
+            throw new InvalidOperationException("Plugin type not found!");
+
+        return type;
+    }
+
+    private async Task OpenEditModal()
+    {
+        EditModalOpen = true;
+        await Task.CompletedTask;
+    }
+
+    private async Task OnEditCancel()
+    {
+        EditModalOpen = false;
+        await Task.CompletedTask;
+    }
+
+    private async Task OnEditSubmit()
+    {
+        EditModalOpen = false;
 
-        queryParams.Add($"redirectTo={redirectTo}");
+        // Should change settings to re render plugins (OnInitalized will be called again)
```

Comment:
@pournasserian I think it would be better to call Settings.GetById API here. 
(plugin developer can change settings of plugin in plugin's onSubmit, we need to get latest settings here)